frontend/packages/console-shared/src/hooks/perspective-utils.ts

```diff
@@ -0,0 +1,160 @@
+      })
+      .catch((e) => {
+        // no update when a missing check fails
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+      }));
+
+  return hasAccess;
+};
+
+export const usePerspectives = (): LoadedExtension<PerspectiveExtension>[] => {
+  const perspectiveExtensions = useExtensions<PerspectiveExtension>(isPerspective);
+  const [results, setResults] = React.useState<Record<string, boolean>>(() => {
+    let obj: Record<string, boolean> = {};
+    if (!window.SERVER_FLAGS.perspectives) {
+      obj = perspectiveExtensions.reduce(
+        (acc: Record<string, boolean>, ex: LoadedExtension<PerspectiveExtension>) => {
+          acc[ex.properties.id] = true;
+          return acc;
+        },
+        {},
+      );
+    } else {
+      const perspectives: Perspective[] = JSON.parse(window.SERVER_FLAGS.perspectives);
+      obj = perspectiveExtensions.reduce((acc, perspectiveExtension) => {
+        const perspective = perspectives?.find((p) => p.id === perspectiveExtension.properties.id);
+
+        if (
+          !perspective?.visibility?.state ||
+          perspective.visibility.state === PerspectiveVisibilityState.Enabled
+        ) {
+          acc[perspectiveExtension.properties.id] = true;
+        } else if (perspective?.visibility?.state === PerspectiveVisibilityState.Disabled) {
+          acc[perspectiveExtension.properties.id] = false;
+        }
+        return acc;
+      }, {});
+    }
+    return obj;
+  });
+
+  const handleResults = React.useCallback(
+    (newResult: string, state: boolean) => {
+      setResults((oldResults: Record<string, boolean>) => {
+        return {
+          ...oldResults,
+          [newResult]: state,
+        };
+      });
+    },
+    [setResults],
+  );
+  React.useEffect(() => {
+    if (window.SERVER_FLAGS.perspectives) {
+      const perspectives: Perspective[] = JSON.parse(window.SERVER_FLAGS.perspectives);
+      perspectiveExtensions.forEach((perspectiveExtension) => {
+        const perspective = perspectives?.find((p) => p.id === perspectiveExtension.properties.id);
+
+        if (
+          !perspective ||
+          !perspective.visibility ||
+          perspective.visibility.state === PerspectiveVisibilityState.Enabled
+        ) {
+          handleResults(perspectiveExtension.properties.id, true);
+        } else if (
+          perspective?.visibility?.state === PerspectiveVisibilityState.AccessReview &&
+          perspective?.visibility?.accessReview &&
+          Object.keys(perspective?.visibility?.accessReview)?.length > 0
+        ) {
+          hasReviewAccess(perspective?.visibility?.accessReview)
+            .then((res) => {
+              if (res) {
+                handleResults(perspectiveExtension.properties.id, true);
+              } else {
+                handleResults(perspectiveExtension.properties.id, false);
+              }
+            })
```

Comment:
Just a nit, you checking already that accessReview is defined and `hasReviewAccess` returns a boolean, so you can just use it:

```suggestion
          perspective?.visibility?.accessReview &&
          Object.keys(perspective?.visibility?.accessReview)?.length > 0
        ) {
          hasReviewAccess(perspective.visibility.accessReview)
            .then((result) => {
              handleResults(perspectiveExtension.properties.id, result);
            })
```